protocols/imap/src/main/java/org/apache/james/imap/decode/ImapRequestLineReader.java

```diff
@@ -369,22 +369,22 @@ public String consumeLiteral(Charset charset) throws DecodingException {
         if (charset == null) {
             return consumeLiteral(US_ASCII);
         } else {
-            try (FastByteArrayOutputStream out = new FastByteArrayOutputStream();
-                 InputStream in = consumeLiteral(false)) {
-                IOUtils.copy(in, out, 2048);
-                byte[] bytes = out.toByteArray();
-                ByteBuffer buffer = ByteBuffer.wrap(bytes);
+            ImmutablePair<Integer, InputStream> literal = consumeLiteral(false);
+            try (InputStream in = literal.right) {
+                Integer size = literal.left;
+                byte[] data = IOUtils.readFully(in, size);
```

Comment:
I know this is orthogonal to your code change, but shouldn't we consider using a DeferredFileOutputStream here as the literal can be big and screw our heap?

Reply:
We could. Some subclasses handle a size limit, some don't. I would not be a refactoring anymore (:

Reply:
I know. I'm just making a remark for future changes :-)